api/envoy/extensions/load_balancing_policies/wrr_locality/v3/wrr_locality.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package envoy.extensions.load_balancing_policies.wrr_locality.v3;
+
+import "envoy/config/cluster/v3/cluster.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.load_balancing_policies.wrr_locality.v3";
+option java_outer_classname = "WrrLocalityProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/load_balancing_policies/wrr_locality/v3;wrr_localityv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Weighted Round Robin Locality-Picking Load Balancing Policy]
+
+// Configuration for the wrr_locality LB policy. See the :ref:`load balancing architecture overview
+// <arch_overview_load_balancing_types>` for more information.
+// [#extension: envoy.clusters.lb_policy]
+message WrrLocality {
+  // The child LB policy to create for endpoint-picking within the chosen locality.
```

Comment:
Where does this recursion bottom-out?

Reply:
It's not really recursion, since it's not going to call back into the same policy; it calls into a different policy that is configured as the child.

The load balancing logic will end when it gets to a policy that returns a result instead of delegating to another child.  So it depends entirely on the configured set of policies.

Reply:
Oh, I see, this is only for locality. Should we have a hierarchy in API distinguishing locality vs leaf?

Reply:
I think it's actually better for the API to be less opinionated than it already is, not more, as per discussion in #5598.  Our experience in gRPC is that we get a lot of flexibility by allowing the LB policies themselves to be organized however they want, rather than by trying to enforce some top-down organizational approach onto them.

Reply:
Makes sense.